### DISCUSSION TITLE
Need help on Input Remapper

### DISCUSSION THREAD

**Discussion by 81ginner** [sezanzeb/input-remapper#950] [Q&A]
This is my first time here. I didn't see anybody else asking exactly the same questions, so here goes:

**Question ONE**

I want to remap the down or 2 button on my numeric keypad to a common combination, control minus. And the up or 8 button to control plus.
These are the combos that zoom into your browser or zoom out. Works on firefox and chrome.

control_L+key_- doesn't work.
control_L + key_- doesn't work.
I also tried key(-), control_L+-, and other combinations. I have been able to remap control_L+alphanumerics such as ctrl+a, ctrl+w, but it is just the - and = that I wasn't able to.
Since I don't actually understand the logic and syntax of programming language, I am just aping the other examples I see online, sorry.

**QUESTION TWO**

According to the info that I've seen, input remapper takes whatever input you actually type on your keyboard into the computer. So theoretically, it is ok to reprogram the numeric keypad. And when you actually want the numbers, to just press numlock which should send numbers to the computer and therefore escape input remapper's remapping.
But I find that whether I press numlock or not, the remapped numeric keypad still does the same thing.

So how does one remap the numeric keypad when numlock is off, and ensure that when numlock is on, input remapper does not remap the numbers?

**QUESTION THREE**

I used to use Autohotkey on Windows. By pressing combinations like Windows key and and alphanumeric like U, I could open http://www.youtube.com. Or I could open a designated folder on my computer. This is obviously very beneficial when you have a folder that is nestled inside so many other folders and you would have to laboriously click through everything to get there.

Is it possible to do this on input remapper?

**Comment by jonasBoss** [comment]
1. for plus and minus you will need to use the actual key-codes: `KEY_KPMINUS` and `KEY_KPPLUS` so basically `Control_L+KEY_KPPLUS`
2. the numlock does not change what the keyboard sends to your computer. So, this is not easy especially because there is currently no god way to see what the numlock status is from within inputremapper. This is probably possible with macros.
3. this is outside the scope of InputRemapper. Almost all linux desktops provide a way to add custom shortcuts to execute commands and shell-scripts, so you can do it that way. There is probably a system-settings menu where you can create and change shortcuts.

**Reply by jonasBoss** [reply]
so ... I just played around with a few macros and came up with the following three mappings:

1. map your num-lock key to:
```
key_down(Num_Lock)
.if_tap(
	if_eq($num, 1, set(num, 0), set(num, 1))
)
.hold()
.key_up(Num_Lock)
```
this will not change your numlock key but keep track of it internally. It could happen that InputRemapper thinks your numlock key is active when it is not, in that case you want to press it long and it will switch what InputRemapper thinks.

2. Map your keypad-8 key to:
```
if_eq(
	$num, 1, 
	hold_keys(KEY_KP8),
	hold_keys(Control_L, KEY_KPPLUS)
)
```
this either sends the 8 key or the zoom macro

3. the keypad-2 key:
```
if_eq(
	$num, 1, 
	hold_keys(KEY_KP2),
	hold_keys(Control_L, KEY_KPMINUS)
)
```
